src/connector/src/source/filesystem/opendal_source/posix_fs_source.rs

```diff
@@ -0,0 +1,54 @@
+// Copyright 2024 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::marker::PhantomData;
+
+use anyhow::Context;
+use opendal::layers::{LoggingLayer, RetryLayer};
+use opendal::services::Fs;
+use opendal::Operator;
+
+use super::opendal_enumerator::OpendalEnumerator;
+use super::{OpendalSource, PosixFsProperties};
+
+impl<Src: OpendalSource> OpendalEnumerator<Src> {
+    /// create opendal posix fs source.
+    pub fn new_posix_fs_source(posix_fs_properties: PosixFsProperties) -> anyhow::Result<Self> {
+        // Create Fs builder.
+        let mut builder = Fs::default();
+
+        builder.root(&posix_fs_properties.root);
+
+        let op: Operator = Operator::new(builder)?
+            .layer(LoggingLayer::default())
+            .layer(RetryLayer::default())
+            .finish();
+
+        let (prefix, matcher) = if let Some(pattern) = posix_fs_properties.match_pattern.as_ref() {
+            // TODO(Kexiang): Currently, FsListnenr in opendal does not support a prefix. (Seems a bug in opendal)
+            // So we assign prefix to empty string.
```

Comment:
so we can deny the prefix in props?

Reply:
If this fs source is only used in testing, it's ok to ignore the miss of prefix list.
BTW, I'm not aware that OpenDAL Fs engine does not support prefix list. cc @Xuanwo

Reply:
The `prefix` is not in props now. In other implemenataion, it's extracted from the `match_pattern`(eg, the prefix of `abc*.csv` is `abc`) and used for filtering out files with a different prefix under the directory/bucket. Since Opendal does not support this listing with `prefix` correctly, we simply set the prefix to empty. If the problem in opendal is solved, we can then generate the prefix here and return it, it benefits to performance.

Reply:
> BTW, I'm not aware that OpenDAL Fs engine does not support prefix list. cc @Xuanwo

opendal 0.44 should fix it.

Reply:
Join the conversation on adding glob support to opendal – your input would be valuable: https://github.com/apache/incubator-opendal/issues/3500